app/pages/cif/project-revision/[projectRevision]/view.tsx

```diff
@@ -97,6 +200,15 @@ export function ProjectRevisionView({
   );
   // Growthbook - amendments
   if (!useShowGrowthbookFeature("amendments")) return null;
+
+  // filtering to show only the amendment statuses that are allowed to be selected based on the revision type
+  const filteredRevisionStatuses = allRevisionStatuses.edges.filter(
+    ({ node }) =>
+      projectRevision.revisionType === "Amendment"
+        ? true
+        : !node.isAmendmentSpecific
+  );
```

Comment:
no need to iterate over the statuses if the revision type is "Amendment", something like that might work better:

```suggestion
  const filteredRevisionStatuses = 
    projectRevision.revisionType === "Amendment" ? 
        allRevisionStatuses.edges :
        allRevisionStatuses.edges.filter(({ node }) => !node.isAmendmentSpecific );
```